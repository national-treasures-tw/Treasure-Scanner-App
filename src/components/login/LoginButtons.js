// @flow
import React from 'react'
import {
  View,
  Text,
  Image,
  TouchableOpacity
} from 'react-native'
import { Images } from '../Themes/'
import styles from './styles/LoginButtonsStyle'

const LoginButtons = (props) => (
  <View style={styles.container}>
    {
      <View style={styles.row}>
        <TouchableOpacity style={styles.fbLoginButton} onPress={props.onFBLoginPressed}>
          <Image source={Images.facebookLogo} style={styles.facebookF} resizeMode='contain' />
          <Text style={styles.fbText}>LOGIN WITH Facebook</Text>
        </TouchableOpacity>
      </View>
    }

    <View style={styles.row}>
      <TouchableOpacity style={styles.loginButton} onPress={props.onLoginButtonPressed}>
        <Text>
          <Text style={styles.loginText}>{'Login'.toUpperCase()} `}</Text>
          <Text style={styles.orText}>{'or'.toUpperCase()} `}</Text>
          <Text style={styles.registerText}>{'Register'.toUpperCase()}</Text>
        </Text>
      </TouchableOpacity>
    </View>

  </View>
)

export default LoginButtons
