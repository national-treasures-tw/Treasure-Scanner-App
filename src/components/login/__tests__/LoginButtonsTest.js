import React from 'react'
import LoginButtons from '../LoginButtons'
import renderer from 'react-test-renderer'

let wrapper
// let x = 0

beforeEach(() => {
  wrapper = renderer.create(<LoginButtons />).toJSON()
})

// afterEach(() => { x = 0 })

test('component exists', () => {
  expect(wrapper).toMatchSnapshot()
})

// test('has the login button', () => {
//   const text = wrapper.find('View > View > TouchableOpacity > Text').first()
//   expect(text.length).toBe(1)
// })

// test('login button has the correct text', () => {
//   const login = wrapper.find('View > View > TouchableOpacity > Text > Text').first().props().children
//   expect(login).toBe(`${I18n.t('login').toUpperCase()} `)

//   const _or = wrapper.find('View > View > TouchableOpacity > Text > Text').at(1).props().children
//   expect(_or).toBe(`${I18n.t('or').toUpperCase()} `)

//   const register = wrapper.find('View > View > TouchableOpacity > Text > Text').last().props().children
//   expect(register).toBe(I18n.t('register').toUpperCase())
// })

// test('facebook button works', () => {
//   const onFBLoginPressed = () => ++x
//   wrapper.setProps({ onFBLoginPressed })
//   const button = wrapper.find('View > View > TouchableOpacity').first()
//   button.simulate('press')
//   expect(x).toBe(1)
// })

// test('login button works', () => {
//   const onLoginButtonPressed = () => ++x
//   wrapper.setProps({ onLoginButtonPressed })
//   const button = wrapper.find('View > View > TouchableOpacity').last()
//   button.simulate('press')
//   expect(x).toBe(1)
// })
