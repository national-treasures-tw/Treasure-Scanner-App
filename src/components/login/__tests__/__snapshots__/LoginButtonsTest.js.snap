// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component exists 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3b5998",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 0.3,
            "height": 20,
            "width": 15,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "flex": 2,
            "fontFamily": "Futura",
            "fontSize": 17,
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        LOGIN WITH Facebook
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "opacity": 1,
          "padding": 6,
          "paddingHorizontal": 40,
          "paddingVertical": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e78720",
              "fontFamily": "Futura",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          LOGIN
           \`}
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e78720",
              "fontFamily": "Futura",
              "fontSize": 17,
              "textAlign": "center",
            }
          }
        >
          OR
           \`}
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e78720",
              "fontFamily": "Futura",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          REGISTER
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
